Added a panel display for cards with no proxies

jpantorilla opened the pull request:
Apart from the panels that are shown when the `Hide Overlay Card Text` are turned on, I've also added a new mechanism that will do some actions when the client versions are mismatched due to an update. 

For this instance, it is simply to do a forced setting update for the overlay setting. I'm planning on adding a patch note feature to the client in the future and will take advantage of this.

jpantorilla committed:
Added no-proxy panel to display translations for cards that has no proxies.

jpantorilla committed:
Values needed for the no-proxy panel for the existing cards that needs them.

jpantorilla committed:
Added a mechanism to do a forced settings update.

jpantorilla committed:
Fixed wrong setup for forced update.

jpantorilla committed:
Merge branch 'main' into no-proxy-display

jpantorilla committed:
updated card definitions and translations for the newly added BD cards

daniel-k-taylor commented on scenes/game/card_base.gd:
```diff
@@ -34,6 +34,8 @@ const CardBaseScene = preload("res://scenes/game/card_base.tscn")
 @onready var active_skill_name = $OuterMargin/ActiveSkillIndicator/PanelContainer/HBoxContainer/SkillContainer/ActiveSkillName
 @onready var targeted_damage_indicator = $OuterMargin/TargetedDamageIndicator
 @onready var targeted_down_indicator = $OuterMargin/TargetedDownIndicator
+@onready var no_proxy_panel = $PanelContainer
```
I'm not a huge fan of "no_proxy" as a descriptive name for what this is.
Maybe popout_info_panel or something?

daniel-k-taylor commented on scenes/game/card_base.gd:
```diff
@@ -60,6 +62,7 @@ var target_position = Vector2.ZERO
 var position_start_time = PositionTime
 var _destroy_on_move_completion = false
 var proxy_card_loaded = false
+var show_no_proxy_panel = false
```
if you change the var names, update this too

daniel-k-taylor commented on globals/global_settings.gd:
```diff
@@ -31,14 +32,15 @@ const user_settings_file = "user://settings.json"
 var user_settings = {
 	GameSound: true,
-	HideEnglishCardText: false,
+	HideEnglishCardText: true,
```
If we're messing with this setting, I'd almost prefer to change it to 2 settings:
Something like "ShowCardTextOnCard" and "HideCardTextForProxies".
With the defaults being False (since you're adding this new thing) and True.
Toggling ShowCardTextOnCard would force the card text to always show on the card.
If HideCardTextForProxies is true, then it shows up even if there's a proxy card, if it's false, it only shows up if there is no proxy card.
UseEnProxies should probably also be renamed to like UseProxies or something.

Thoughts?

jpantorilla replied:
I'm a bit confused with the settings you're proposing. Which card texts are we talking about here? Let's use these terms for clarity: **``overlay``** for the generated text that is within the card; **``panel``** for this feature.

daniel-k-taylor replied:
In that case, I guess it makes more sense with these names:
ShowCardOverlayText
HideCardOverlayTextForProxies

ShowCardOverlayText = true - the overlay always shows up.  False - it never shows up.
HideCardOverlayTextForProxies = true - hides the overlay text for proxy cards even if ShowCardOverlayText=true

The panel feature will show up if ShowCardOverlayText is false, which we can set as the default.

Does that make sense?

jpantorilla replied:
> HideCardOverlayTextForProxies

I don't see a reason for this and it just overcomplicates things. What happens if ``hide_for_proxies`` is ``false`` and ``show_overlay`` is ``true``. That would make it so that the overlay will cover the card even if the image is fine. That would be bad UX.

I feel that the current settings we have works fine. It only shows the ``overlay/panel`` for cards we have no proxies for that particular language. We'd maybe just have to rename the setting to something better, maybe even add a tooltip.

Actually, I'm open to ditching adding the translations and use the generated text instead. That way, we can remove the ``overlay`` and just have the ``panel`` for the presentation.

daniel-k-taylor replied:
show overlay true and hide for proxies false is actually useful for debugging and checking how the strings show up on specific cards.
Also the overlay text is somewhat readable on the board compared to the proxy cards which are completely unreadable without zoom. I could see someone wanting to have them up all the time.

I'm definitely for just having one version of the strings, preferably the generated text.

jpantorilla replied:
```
- Card Info [ ]
   -- Overlay [ ]
   -- Panel [ ]
```
I guess this would be the best setup?

daniel-k-taylor replied:
That's fine

daniel-k-taylor commented on scenes/game/card_base.gd:
```diff
@@ -439,3 +446,104 @@ func _on_button_mouse_exited() -> void:
 func _on_attachment_button_pressed() -> void:
 	view_attachments.emit(_attached_cards)
+
+func _update_no_proxy_panel(card: CardBase) -> void:
```
What are your thoughts on this functionality being contained in its own scene and CardBase just having one of these?
Especially if we think we're going to add significantly more code.
If not, maybe it's fine.

jpantorilla replied:
I agree with separating it to its own scene since ``CardBase`` responsibility here is just to toggle its visibility and supplying the data.

daniel-k-taylor reviewed (changes requested):
I'm a bit hesitant to start adding a ton of full card text for things we should be able to pretty reasonably have in programmatic strings.
Maybe we can just format the programmatic strings better?
Especially for the repeated ones like the BD cards.

jpantorilla committed:
Moved CardInfoPanel into its own scene.

jpantorilla committed:
Introduced new settings for controlling the display for card info overlay and panel.

jpantorilla committed:
Removed unnecessary signal for showing panels.

jpantorilla committed:
Used the generated text instead of the translations.

jpantorilla committed:
Removed unused translation-related stuff.

jpantorilla commented:
Added changes based on our discussions:
- Moved CardInfoPanel into its own scene.
- Added new settings to control what kind of card info will be displayed.
- Used generated text instead of adding translation texts.

<img width="350" alt="screenshot" src="https://github.com/user-attachments/assets/2e815c4c-6f38-44ab-ab17-b84bb81f0cdd">
<img width="350" alt="screenshot" src="https://github.com/user-attachments/assets/b3b4da9d-087f-4519-8560-bd503523fa28">
<img width="350" alt="screenshot" src="https://github.com/user-attachments/assets/e8f13d16-cb10-46b3-bc0f-f17d60822f0d">
<img width="175" alt="screenshot" src="https://github.com/user-attachments/assets/9d8a21ab-7cff-4ddb-b3bf-fffaa6258d9a">

jpantorilla committed:
Removed client-version checking feature as it is unused in this PR.
